core/src/stack/outputs.rs

```diff
@@ -218,22 +222,34 @@ impl ToElements<Felt> for StackOutputs {
 
 impl Serializable for StackOutputs {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
-        debug_assert!(self.stack.len() <= u32::MAX as usize);
-        target.write_u32(self.stack.len() as u32);
+        debug_assert!(self.stack.len() <= MAX_STACK_OUTPUTS_SIZE);
+        target.write_usize(self.stack.len());
         target.write_many(&self.stack);
 
-        debug_assert!(self.overflow_addrs.len() <= u32::MAX as usize);
-        target.write_u32(self.overflow_addrs.len() as u32);
+        debug_assert!(self.overflow_addrs.len() <= MAX_OVERFLOW_ADDRS_NUM);
+        target.write_usize(self.overflow_addrs.len());
         target.write_many(&self.overflow_addrs);
     }
 }
 
 impl Deserializable for StackOutputs {
     fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
-        let count = source.read_u32()?.try_into().expect("u32 must fit in a usize");
+        let count = source.read_usize()?;
+        if count > MAX_STACK_OUTPUTS_SIZE {
+            return Err(DeserializationError::InvalidValue(format!(
+                "Number of values in the output stack can not be more than {}, but {} was found",
```

Comment:
minor nit: I'd say "on the output stack" rather than "in the output stack" (this may be relevant to other places too).